source/Mlos.Python/mlos/Optimizers/unit_tests/TestBayesianOptimizer.py

```diff
@@ -406,6 +406,44 @@ def test_bayesian_optimizer_default_copies_parameters(self):
         print(original_config.experiment_designer_config.fraction_random_suggestions)
         assert original_config.experiment_designer_config.fraction_random_suggestions == .5
 
+
+    def test_registering_multiple_objectives(self):
+
+        input_space = SimpleHypergrid(
+            name='input',
+            dimensions=[
+                ContinuousDimension(name="x_1", min=0, max=10),
+                ContinuousDimension(name="x_2", min=0, max=10)
+            ]
+        )
+
+        output_space = SimpleHypergrid(
+            name='output',
+            dimensions=[
+                ContinuousDimension(name="y_1", min=0, max=10),
+                ContinuousDimension(name="y_2", min=0, max=10)
+            ]
+        )
+
+        optimization_problem = OptimizationProblem(
+            parameter_space=input_space,
+            objective_space=output_space,
+            objectives=[Objective(name='y_1', minimize=True)]
+        )
+
+        optimizer = self.bayesian_optimizer_factory.create_local_optimizer(
+            optimization_problem=optimization_problem
+        )
+
+        for _ in range(100):
+            input = optimizer.suggest()
+            output = Point(y_1=input.x_1, y_2=input.x_2)
+
+            optimizer.register(input.to_dataframe(), output.to_dataframe())

```

Comment:
Test what happens if you submit only one of the required outputs. Also test what happens if you submit extras. Make sure such situations are handled gracefully. Also make sure that you can retrieve all observations.